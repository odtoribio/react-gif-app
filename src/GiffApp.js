import React,  { useState } from 'react'
import { AddCategoryButton } from './components/AddCategoryButton'
import { GiffGrid } from './components/GiffGrid'

export const GiffApp = ({ defaultCategories=[]}) => {

  const [categories, setCategories] = useState(defaultCategories)

  const addNewCategory = (value) => {
    setCategories([value, ...categories])
  }

  return (
    <>
      <h2>Giff Expert App</h2>
      <AddCategoryButton addCategory={ addNewCategory } />
      <hr/>
      <ol>
      {
        categories.map( (category ) => (
          <GiffGrid key={ category } category={ category } />
        ))
      }
      </ol>
    </>
  )
}
