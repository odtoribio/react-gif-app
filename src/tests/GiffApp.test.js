import { GiffApp } from '../GiffApp'
import { shallow } from 'enzyme'

describe('Test GiffApp component', () => {

  test('should render GiffApp component', () => {
    const GiffAppComponent = shallow(<GiffApp />);
    expect(GiffAppComponent).toMatchSnapshot()
  })

  test('should render a list of categories', () => {
    const categories = ['one punch, goku']
    const GiffAppComponent = shallow(<GiffApp defaultCategories={ categories } />);

    expect(GiffAppComponent).toMatchSnapshot()
    expect(GiffAppComponent.find('GiffGrid').length).toBe(categories.length)
  })
  
})

