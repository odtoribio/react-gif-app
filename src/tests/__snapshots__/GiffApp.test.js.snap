// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GiffApp component should render GiffApp component 1`] = `
<Fragment>
  <h2>
    Giff Expert App
  </h2>
  <AddCategoryButton
    addCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test GiffApp component should render a list of categories 1`] = `
<Fragment>
  <h2>
    Giff Expert App
  </h2>
  <AddCategoryButton
    addCategory={[Function]}
  />
  <hr />
  <ol>
    <GiffGrid
      category="one punch, goku"
      key="one punch, goku"
    />
  </ol>
</Fragment>
`;
